### ISSUE TITLE
Combiner with weigths

### ISSUE THREAD

**Issue by crawfordsm** [astropy/ccdproc#621]
Line 133 in Combiner.py compares two shapes and should only be doing it on the first element of each shape rather than the arrays:
   
                     if value.shape == self.data_arr.data.shape:

**Comment by MSeifert04** [comment]
Interesting. 

However currently broadcasting fails if the weights don't have the same shape as the data. So probably this was on purpose.

```
from ccdproc import Combiner, CCDData
import numpy as np

ccds = [CCDData(np.random.random((10, 10)), unit='adu') for _ in range(10)]

c = Combiner(ccds)
c._weights = np.random.random((1, 10, 10))  # or np.random.random((10, 10))
c.average_combine()
```

That throws `TypeError: 1D weights expected when shapes of a and weights differ.`. We probably need to do some reshaping if it's not equal to the stacked size but equal to the individual array sizes...

**Comment by MSeifert04** [comment]
Interesting, [you actually added a test case](https://github.com/astropy/ccdproc/blame/master/ccdproc/tests/test_combiner.py#L100-L104) that weights with the same shape as one ccd will raise an exception. I wonder if this was intention?

**Comment by crawfordsm** [comment]
Closed with #670
